public/components/incontext_insight/index.tsx

```diff
@@ -174,17 +258,86 @@ export const IncontextInsight = ({ children }: IncontextInsightProps) => {
     </EuiPopoverFooter>
   );
 
-  const GeneratePopoverBody: React.FC<{}> = ({}) => (
-    <EuiButton onClick={() => toasts.addDanger('To be implemented...')}>Generate summary</EuiButton>
-  );
+  const GeneratePopoverBody: React.FC<{ incontextInsight: IncontextInsightInput }> = ({
```

Comment:
Can we separate `GeneratePopoverBody` outside as an isolate component? It take about 120 lines. Prefer move `onGenerateSummary`, `onChatContinuation` and `isLlmResponded`, `isLoading` state in the same component too.

Reply:
Now it's an independent component in a different file - generate_popover_body.tsx